src/libs/actions/OnyxUpdateManager.ts

```diff
@@ -41,9 +41,14 @@ export default () => {
     Onyx.connect({
         key: ONYXKEYS.ONYX_UPDATES_FROM_SERVER,
         callback: (value) => {
-            // When the OpenApp command hasn't finished yet, we should not process any updates.
+            // When value is set to null (which we do once we finish processing this method), we can just return early.
+            // Alternatively, if isLoadingApp is positive, that means that OpenApp command hasn't finished yet, so we
+            // should not process any updates that are being set.
             if (!value || isLoadingApp) {
                 if (isLoadingApp) {
+                    // Some of the places that set ONYX_UPDATES_FROM_SERVER also stop the sequential queue. If we reached
```

Comment:
It is weird to see `isLoadingApp` in both `if` statements here. I think it would make the code more readable to turn this into two separate early returns. Then the comments might make more sense too.

Reply:
There are only three places in the code which set `ONYX_UPDATES_FROM_SERVER`.

- Two of them call `SequentialQueue.unpause();` right after which is probably fine and good
- [This](https://github.com/Expensify/App/blob/ad1a1a6860cc5126e36d59a0837b54ff33774572/src/libs/actions/OnyxUpdates.ts#L123) is the only one that doesn't... but it is also not pausing the queue anywhere close to it, so it doesn't appear like your comment is accurate.

Maybe a better solution is to put the `SequentialQueue.unpause();` right after [this](https://github.com/Expensify/App/blob/ad1a1a6860cc5126e36d59a0837b54ff33774572/src/libs/actions/OnyxUpdates.ts#L123) just like the other occurrences? It would make more sense to put it there but I am not sure if it achieves the same result.